main.cpp

```diff
@@ -53,103 +53,32 @@ Project 4: Part 4 / 9
  You will need to use Forward Declaration and out-of-class definitions to complete this.
  */
 
-
+struct FloatType;
+struct DoubleType;
+struct IntType;
 
 struct Point
 {
+    Point(float x, float y);
+    explicit Point(FloatType& x, FloatType& y);
+    explicit Point(DoubleType& x, DoubleType& y);
+    explicit Point(IntType& x, IntType& y);
+    
     Point& multiply(float m)
     {
-        x *= m;
-        y *= m;
+        mx *= m;
+        my *= m;
         return *this;
     }
+    Point& multiply(FloatType& m);
+    Point& multiply(DoubleType& m);
+    Point& multiply(IntType& m);
+    void toString();
+
 private:
-    float x{0}, y{0};
+    float mx{0}, my{0};
```

Comment:
these did not need their name changed. 
Put it back to what it was.
Change the function arguments that are causing the "overshadow" warnings. 
You learned this in Project 3 Part 2:
```
 2) provide implementations for the member functions you declared in your 10 user-defined types from the previous video outside of your UDT definitions.
    If you have 'unused parameter' warnings, you aren't using one of your function parameters in your implementation.
    Solution: use the parameter in your implementation.

    If you have 'shadows a field of <classname>' warnings, a local variable in the function has the same name as a class member.  
    This local variable could also be a function parameter with the same name as the class member.
    Solution: change the local variable's name so it is different from the class member variable's name.
```

- [ ] revise